SimulationAgent/Agent.cs

```diff
@@ -59,17 +63,27 @@ public async Task RunAsync()
                 {
                     this.log.Debug("------ Checking for simulation changes ------");
 
-                    var newSimulation = (await this.simulations.GetListAsync()).FirstOrDefault();
-                    this.log.Debug("Simulation loaded", () => new { newSimulation });
+                    var simulationList = await this.simulations.GetListAsync();
+
+                    // currently we support only 1 running simulation so the result should return only 1 item
+                    var runningSimulation = simulationList.FirstOrDefault(s => s.ShouldBeRunning());
+
+                    this.log.Debug("Simulation loaded", () => new { runningSimulation });
 
                     // if the simulation is removed from storage & we're running stop simulation.
-                    this.CheckForDeletedSimulation(newSimulation);
+                    var id = this.simulation?.Id;
+                    var prevSimulation = simulationList.FirstOrDefault(s => s.Id == id);
+                    this.CheckForDeletedSimulation(prevSimulation);
```

Comment:
this code is going to be rewritten, however: if not too hard, move the added logic inside one of the 4 methods (CheckForDeletedSimulation, CheckForChangedSimulation, CheckForNewSimulation, etc.) or add a new one.  This method used to be very long, with a lot of cases to handle, and we split it into those 4 private methods.   Anyway, as I said, this will be rewritten, so consider it just an observation

Reply:
I will leave this as is for now.

Reply:
:white_check_mark: